// vue-use-paginator/dist/vue-use-paginator.es.js
import { ref as m, computed as h, watch as z } from "vue";
const B = {
  page: 1,
  pageSize: 5,
  numItems: 0,
  numButtons: 5
}, b = (e) => {
  const s = { ...B, ...e }, o = m(s.page), t = m(s.pageSize), a = m(s.numItems), u = m(s.numButtons), l = h({
    get: () => o.value,
    set: (p) => {
      p > 0 && p <= n.value && (o.value = p);
    }
  }), n = h(() => Math.ceil(a.value / t.value));
  z([t, a], () => {
    l.value > n.value && (l.value = n.value);
  });
  const c = h(() => {
    const p = (l.value - 1) * t.value, S = Math.min(p + t.value, a.value);
    return [p, S];
  }), i = h(() => l.value > 1), v = h(() => l.value < n.value), f = () => (i.value && l.value--, l.value), d = () => (v.value && l.value++, l.value), M = () => (l.value = 1, 1), g = () => (l.value = n.value, n.value), x = h(() => n.value <= u.value ? I(l.value, n.value) : N(l.value, n.value, u.value));
  return {
    page: l,
    pageSize: t,
    numItems: a,
    numButtons: u,
    numPages: n,
    slice: c,
    hasPrev: i,
    hasNext: v,
    goPrev: f,
    goNext: d,
    goStart: M,
    goEnd: g,
    buttons: x
  };
};
function I(e, s) {
  return r(1, e).concat({ page: e, active: !0, ellipsis: !1 }).concat(r(e + 1, s + 1));
}
function N(e, s, o) {
  let t = 0, a = 0;
  const u = Math.floor(o / 2);
  e <= u ? (t = e - 1, a = o - t - 1) : e >= s - u ? (a = s - e, t = o - a - 1) : o % 2 === 0 ? (t = u - 1, a = u) : (t = u, a = u);
  const l = e - 1, n = s - e;
  let c = [];
  if (l <= t)
    c = r(1, l + 1);
  else if (t > 0) {
    const i = t === 2 ? 2 : Math.ceil(t / 2), v = t - i;
    let f = Math.ceil(l / 2);
    l % 2 === 0 && f++, c = r(1, i).concat({ page: f, active: !1, ellipsis: !0 }).concat(r(e - v, e));
  }
  if (c.push({ page: e, active: !0, ellipsis: !1 }), n <= a)
    c = c.concat(r(e + 1, s + 1));
  else if (a > 0) {
    const i = a === 2 ? 1 : Math.floor(a / 2) + 1, v = a - i, f = e + Math.ceil(n / 2);
    c = c.concat(r(e + 1, e + i)).concat({ page: f, active: !1, ellipsis: !0 }).concat(r(s - v + 1, s + 1));
  }
  return c;
}
function r(e, s) {
  const o = [];
  for (let t = e; t < s; t++)
    o.push({ page: t, active: !1, ellipsis: !1 });
  return o;
}
export {
  b as default
};
